src/reconcile/rec_visibility.c

```diff
@@ -322,6 +322,16 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         return (0);
     }
 
+    /*
+     * We expect the page to be clean after reconciliation when closing a tree. If there are newer
+     * updates, abort eviction and retry.
```

Comment:
Where is the retry that you reference here? I thought we would give up trying to evict the file if this fails with EBUSY.

Reply:
The caller of `__wt_eict_file`. Here's the comment in evict_file.c, which describe the same case.
```
* Eviction can also fail because an update cannot be written. If sessions have disjoint
* sets of files open, updates in a no-longer-referenced file may not yet be globally
* visible, and the write will fail with EBUSY. Our caller handles that error, retrying
 * later.
```